src/constructs/aws/StaticWebsite.ts

```diff
@@ -17,7 +19,18 @@ export class StaticWebsite extends StaticWebsiteAbstract {
         protected readonly configuration: CommonStaticWebsiteConfiguration,
         protected readonly provider: AwsProvider
     ) {
-        super(scope, id, configuration, provider);
+        // we need to build specific bucket props to pass to the abstract static website constructor
+        const bucketProps: BucketProps = {
+            // Enable static website hosting
+            websiteIndexDocument: "index.html",
+            websiteErrorDocument: configuration.errorPage,
+            // public read access is required when enabling static website hosting
+            publicReadAccess: true,
+            // For a static website, the content is code that should be versioned elsewhere
+            removalPolicy: RemovalPolicy.DESTROY,
+        };
+
+        super(scope, id, configuration, provider, bucketProps);
```

Comment:
`super` should always be called first in any class implementing an abstract class in order to avoid anti-pattern in composition. If there are 2 different bucket configuration for SPA and static website, bucket creation should be the scope of those specific construct. Cloudfront configuration common resources should remain within the abstract `StaticWebsiteAbstract` construct.